src/depthai_wrapper.cpp

```diff
@@ -0,0 +1,133 @@
+#include "depthai/depthai_wrapper.hpp"
+#include <fstream>
+#include <sstream>
+
+#define WARNING "\033[1;5;31m" // Adds color to error statements
+#define ENDC "\033[0m"
+
+namespace DepthAI {
+
+const std::unordered_map<int, int> DepthAI::width_to_height_map_ = { { 720, 1280 }, { 800, 1280 }, { 400, 640 } };
+
+DepthAI::DepthAI(const std::string& usb_device, const std::string& config_file, bool usb2_mode)
+    : Device(usb_device, usb2_mode)
+{
+    // opening config file
+    std::ifstream file(config_file);
+    std::ostringstream file_stream;
+    if (file) {
+        file_stream << file.rdbuf();
+    } else {
+        std::cerr << WARNING + config_file + " file  is not found" ENDC << std::endl;
+        abort();
+    }
+    std::string config_str = file_stream.str();
+    // creating pipeline using config and saving the json of config.
+    pipeline_ = create_pipeline(config_str);
+    config_json_ = nlohmann::json::parse(config_str);
+
+    // extract resolution of the frames and create cv::Mat placeholders
+    set_resolution();
+    create_frame_holders();
+}
+
+void DepthAI::set_resolution()
+{
+    // if camera is defined check if resolution of stereo or rgb is defined. else set to default
+    if (config_json_.contains("camera")) {
+        auto& camera_conf_obj = config_json_.at("camera");
+        /*
+         * if rgb res is set in config_json_ fetch it to set 
+         * the height of the image else set the default
+         */
+        if (camera_conf_obj.contains("rgb")) {
+            auto& rgb_camera_conf_obj = camera_conf_obj.at("rgb");
+            rgb_height_ = rgb_camera_conf_obj.at("resolution_h").get<int32_t>();
+            // _rgb_width = width_to_height_map_[_rgb_height];
+        } else {
+            rgb_height_ = 3040;
+            rgb_width_ = 4056;
+        }
+        /*
+         * if stereo camera res is set in config_json_ fetch it to set 
+         * the height of the image else set the default
+         */
+        if (camera_conf_obj.contains("mono")) {
+            auto& mono_camera_conf_obj = camera_conf_obj.at("mono");
+            mono_height_ = mono_camera_conf_obj.at("resolution_h").get<int32_t>();
+            auto it = width_to_height_map_.find(mono_height_);
+            mono_width_ = it->second;
```

Comment:
Why are you searching the map with height as key? The keys are widths right? Similarly, why is `mono_width_ = it->second`? It should be `it->first` right?